### PR TITLE
Bash completion

### PR DESCRIPTION
**Pull request by tartansandal**
Checklist

- [ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ x] Source is from official source
- [ x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ x] If static libraries are linked in, the license of the static library is packaged.
- [ x] Build number is 0
- [ x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by jan-janssen** [comment]
Is this for Linux only? Then add `skip: true  # [not linux]` to the build section

File: recipes/bash-completion/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - double-load-trap.patch
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - autoconf
+    - automake
+
```
Comment (scopatz) [inline recipes/bash-completion/meta.yaml]: This has a runtime dependency on bash

Comment (beckermr) [reply recipes/bash-completion/meta.yaml]: So we don't allow users to install this and use it against their own bash? Or does it not work at all in that case?

**Comment by tartansandal** [comment]
> Is this for Linux only? Then add `skip: true # [not linux]` to the build section

In theory it should work on MacOS and CygWin as well as Linux, though I've only been able to test on Linux so far.  I was being cheeky and using your CI to get some feedback.  

Is it okay if I set it to `skip: true # [win]` so we can work on the MacOS issues -- the should not be too hard to solve?

File: recipes/bash-completion/meta.yaml (hunk 1 above)
Comment (tartansandal) [reply recipes/bash-completion/meta.yaml]: It is intended for use against the user's own bash, but it should also work with a conda installed version.  It only makes sense for interactive shell and was packaged as a hard dependency for the `conda-bash-completion` code (see https://github.com/conda-forge/staged-recipes/pull/10158).

Does this mean it has a `host` dependency on bash?  (Full disclosure: very new to the whole conda echo system).

**Comment by tartansandal** [comment]
Damn. It looks like macOS has a bash-3.2.57 from way back in 2007.  This library requires at least version 4.1.

File: recipes/bash-completion/README.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
+# A conda recipe for the `bash-completion` package.
```
Comment (beckermr) [inline recipes/bash-completion/README.md]: This file should be removed.

**Comment by beckermr** [comment]
Ugh. Two options. 

1. add bash to the run time deps of the recipe
2. make the library compatible with the older bash

**Comment by tartansandal** [comment]
> 1. add bash to the run time deps of the recipe

I think that would only work if your active shell is a sub-process of a login shell that already add the conda bits to PATH.  Commonly this is the case for a terminal in a desktop Linux, but not for the console or an SSH session. Not sure how this works for, say, iTerm2 on MacOS, or an SSH session for that matter.  From memory you can get a modern bash with `brew`.

> 2. make the library compatible with the older bash

Sadly not really practical. 3000+ lines of shell code. Then we would have to run the tests which depend on `dejagnu` which we would have to package.  

Note the activation scripts have version guards, so the library is a no-op if the bash version is not sufficient.

I'm going to disable the `macOS` build until we can get some input from those users.

**Pull request merged**